File: deps/src/to-todd-coxeter.cpp
#include "libsemigroups_julia.hpp"

#include <cstdint>

#include <libsemigroups/froidure-pin-base.hpp>
#include <libsemigroups/knuth-bendix-class.hpp>
#include <libsemigroups/to-todd-coxeter.hpp>
#include <libsemigroups/types.hpp>
#include <libsemigroups/word-graph.hpp>

namespace libsemigroups_julia {

  void define_to_todd_coxeter(jl::Module& m) {
    m.method("to_todd_coxeter_from_fpb",
             [](libsemigroups::congruence_kind            knd,
                libsemigroups::FroidurePinBase&           fpb,
                libsemigroups::WordGraph<uint32_t> const& wg) {
               return libsemigroups::to<
                   libsemigroups::ToddCoxeter<libsemigroups::word_type>>(
                   knd, fpb, wg);
             });

    m.method("to_todd_coxeter_from_kb",
             [](libsemigroups::congruence_kind                        knd,
                libsemigroups::KnuthBendix<libsemigroups::word_type>& kb) {
               return libsemigroups::to<libsemigroups::ToddCoxeter>(knd, kb);
             });
  }

}  // namespace libsemigroups_julia

File: src/to-todd-coxeter.jl
"""
to-todd-coxeter.jl - conversions to ToddCoxeter
"""

"""
    to(::Type{ToddCoxeter}, kind::congruence_kind, fp::FroidurePin,
       wg::WordGraph) -> ToddCoxeter

Convert a Froidure-Pin graph to a Todd-Coxeter.
"""
to(::Type{ToddCoxeter}, kind::congruence_kind, fp::FroidurePin, wg) =
    LibSemigroups.to_todd_coxeter_from_fpb(kind, fp.cxx_obj, wg)

"""
    to(::Type{ToddCoxeter}, kind::congruence_kind, kb::KnuthBendix) -> ToddCoxeter

Convert a Knuth-Bendix to a Todd-Coxeter.
"""
to(::Type{ToddCoxeter}, kind::congruence_kind, kb::KnuthBendix) =
    LibSemigroups.to_todd_coxeter_from_kb(kind, kb)
